app/models/race.rb

```diff
@@ -32,7 +32,7 @@ def name_with_event
   end
 
   def generate_slug
-    "#{starts_at.year} #{event.name} #{name}".parameterize
+    "#{starts_at&.year} #{event&.name} #{name}".parameterize
```

Comment:
Why the safe-nav here? If anything, those values should _always_ be present before slugging.

Reply:
I was getting that nil class error above. I replaced the `if @race` line in the model with the safe-nav operators. Both resolved the error for the test but I'll keep looking at it. I guess I don't have my mind fully wrapped around what's going on with the error.